File: 0252-meeting-rooms/0252-meeting-rooms.c
"""
 Input: intervals = [[0,30],[5,10],[15,20]]
                     s1  e1  s2 e2  s3   e3
                     
  sort(intervals) 
  for i in range(len(intervals)):
    # if e1 > s2, then false 
    if intervals[i][1] > intervals[i+1][0]:
       rturn False
    
  # e1 < s2
  return True

"""


class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        
        intervals.sort()
        for i in range(len(intervals) - 1):
            if intervals[i][1] > intervals[i+1][0]:
                return False
        return True
        

File: 0252-meeting-rooms/0252-meeting-rooms.go
func canAttendMeetings(intervals [][]int) bool {
    // Sorting function for intervals
    sort.Slice(intervals, func(i, j int) bool {
        if intervals[i][0] == intervals[j][0] {
            return intervals[i][1] < intervals[j][1]
        } else {
            return intervals[i][0] < intervals[j][0]
        }
    })
     
    for i := 0; i + 1 < len(intervals); i++ {
        if (intervals[i][1] > intervals[i+1][0]) {
            return false
        }
    }
    return true
}

// Time complexity: O(nlog(n)), space complexity: O(1)